app/views/layouts/application.html.erb

```diff
@@ -13,15 +13,20 @@
       <%= link_to "Home", "/" %>
       <%= link_to "All Merchants", "/merchants"%>
       <%= link_to "All Items", "/items"%>
-      <%= link_to "Login", "/login" %>
-      <%= link_to "Register", "/users/new", method: :get %>
-      <% if current_admin? %>
-        <%= link_to "Admin Dashboard", "/admin" %>
-        <%= link_to "Users", "/admin/users" %>
-      <% elsif current_merchant? %>
-        <%= link_to "Merchant Dashboard", "/merchant" %>
+      <% if current_user %>
+        <% if current_admin? %>
+          <%= link_to "Admin Dashboard", "/admin" %>
+          <%= link_to "Users", "/admin/users" %>
+        <% elsif current_merchant? %>
+          <%= link_to "Merchant Dashboard", "/merchant" %>
+          <%= link_to "Cart: #{cart.total_items}", "/cart" %>
+        <% end %>
+        <%= link_to "Profile", "/profile" %>
+        <%= link_to "Logout", "/logout" %>
       <% else %>
         <%= link_to "Cart: #{cart.total_items}", "/cart" %>
```

Comment:
I see that the cart is listed twice. on line 22 and 27, which one would be better to keep in your opinion?

Reply:
Yes good question I think both are needed to satisfy the user stories.

If there is no current user there should be a cart for a visitor.

If there is a user and the user is an admin the user story says there should not be a cart link
The way I solved that was by having by putting the cart link twice.